s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
(the latter point is only relevant if we really want the filesystem to be quasi-posix, for instance for later FUSE integration)

OK to make ls with missing bucket an error; I don't think it's an error to list a bucket that contains no keys, though, only specifying a path where the root doesn't exist either,